applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -98,6 +98,17 @@ void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
     // Tensor variable definition
     const Variable<TensorArrayType>& tensor_variable = KratosComponents<Variable<TensorArrayType>>::Get("METRIC_TENSOR_"+std::to_string(TDim)+"D");
 
+    // Setting metric in case not defined
+    if (!nodes_array.begin()->Has(tensor_variable)) {
+        // Declaring auxiliar vector
+        const TensorArrayType aux_zero_vector = ZeroVector(3 * (TDim - 1));
+        #pragma omp parallel for
+        for(int i = 0; i < num_nodes; ++i) {
+            auto it_node = nodes_array.begin() + i;
+            it_node->SetValue(tensor_variable, aux_zero_vector);
```

Comment:
are you sure this works in omp?
If it is not allocated before the data-value-container will throw an error in debug mode, can you please check it?

Reply:
Yes, it works, look at the code of the MetricFastInit:

~~~cpp
template<SizeType TDim>
void MetricFastInit<TDim>::Execute()
{
    KRATOS_TRY;

    const TensorArrayType zero_array(3 * (TDim - 1), 0.0);

    // We iterate over the nodes
    NodesArrayType& nodes_array = mrThisModelPart.Nodes();

    // Tensor variable definition
    const Variable<TensorArrayType>& tensor_variable = KratosComponents<Variable<TensorArrayType>>::Get("METRIC_TENSOR_"+std::to_string(TDim)+"D");

    #pragma omp parallel for
    for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i)
        (nodes_array.begin() + i)->SetValue(tensor_variable, zero_array);

    KRATOS_CATCH("");
}
~~~

Reply:
ok